commit 94fdc1d3b2633df350a7e7c7d74deef401a1c10f (refs/changes/69/1069/1)
Author: Todd Whiteman <todd.whiteman@joyent.com>
Date:   2016-12-05T16:49:04-08:00 (2 years, 10 months ago)
    
    joyent/sdc-docker#103 docker build --force-rm=true should always remove intermediate containers

diff --git a/lib/backends/sdc/build.js b/lib/backends/sdc/build.js
index 1b9c4cf..13949fa 100644
--- a/lib/backends/sdc/build.js
+++ b/lib/backends/sdc/build.js
@@ -507,7 +507,7 @@ function buildImage(opts, callback) {
     function cleanup(err, cb) {
         if (vmUuid) {
             // Cleanup according to the user's cleanup rules.
-            if ((!err && dockerOpts.rm) || (err && dockerOpts.forcerm)) {
+            if ((!err && dockerOpts.rm) || dockerOpts.forcerm) {
                 // Ensure the vm is deleted.
                 log.debug('Removing the build container');
                 deleteVm(vmUuid, function (deleteErr) {
